pkg/controller/kubelet-config/kubelet_config_nodes.go

```diff
@@ -0,0 +1,336 @@
+package kubeletconfig
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/clarketm/json"
+	"github.com/golang/glog"
+	osev1 "github.com/openshift/api/config/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/retry"
+	"k8s.io/kubelet/config/v1beta1"
+
+	mcfgv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	ctrlcommon "github.com/openshift/machine-config-operator/pkg/controller/common"
+	"github.com/openshift/machine-config-operator/pkg/version"
+)
+
+func (ctrl *Controller) nodeConfigWorker() {
+	for ctrl.processNextNodeConfigWorkItem() {
+	}
+}
+
+func (ctrl *Controller) processNextNodeConfigWorkItem() bool {
+	key, quit := ctrl.nodeConfigQueue.Get()
+	if quit {
+		return false
+	}
+	defer ctrl.nodeConfigQueue.Done(key)
+
+	err := ctrl.syncNodeConfigHandler(key.(string))
+	ctrl.handleNodeConfigErr(err, key)
+	return true
+}
+
+func (ctrl *Controller) handleNodeConfigErr(err error, key interface{}) {
+	if err == nil {
+		ctrl.nodeConfigQueue.Forget(key)
+		return
+	}
+
+	if ctrl.nodeConfigQueue.NumRequeues(key) < maxRetries {
+		glog.V(2).Infof("Error syncing node configuration %v: %v", key, err)
+		ctrl.nodeConfigQueue.AddRateLimited(key)
+		return
+	}
+
+	utilruntime.HandleError(err)
+	glog.V(2).Infof("Dropping node config %q out of the queue: %v", key, err)
+	ctrl.nodeConfigQueue.Forget(key)
+	ctrl.nodeConfigQueue.AddAfter(key, 1*time.Minute)
+}
+
+// syncNodeConfigHandler syncs whenever there is a change on the nodes.config.openshift.io resource
+// nodes.config.openshift.io object holds the cluster-wide information about the
+// node specific features such as cgroup modes, workerlatencyprofiles, etc.
+func (ctrl *Controller) syncNodeConfigHandler(key string) error {
+	startTime := time.Now()
+	glog.V(4).Infof("Started syncing nodeConfig handler %q (%v)", key, startTime)
+	defer func() {
+		glog.V(4).Infof("Finished syncing nodeConfig handler %q (%v)", key, time.Since(startTime))
+	}()
+
+	// Fetch the Node
+	nodeConfig, err := ctrl.nodeConfigLister.Get(ctrlcommon.ClusterNodeInstanceName)
+	if errors.IsNotFound(err) {
+		glog.V(2).Infof("Node configuration %v is missing, using default", key)
+		nodeConfig = createNewDefaultNodeconfig()
+	} else if err != nil {
+		glog.V(2).Infof("%v", err)
+		err := fmt.Errorf("could not fetch Node: %v", err)
+		return err
+	}
+	// Find all MachineConfigPools
+	mcpPools, err := ctrl.mcpLister.List(labels.Everything())
+	if err != nil {
+		return err
+	}
+
+	for _, pool := range mcpPools {
+		role := pool.Name
+		// the configuration change will be applied only on the worker nodes.
```

Comment:
Same question as above. Does the nodes.config.openshift.io resources only apply to workers? Or any custom pools?

Reply:
custom pools inherit from worker pool so once a new worker mc is generated, they'd get this update too no? (unless I'm completely missing something here)

Reply:
Ofc that said, then we should only be checking for the worker pool and not create the impression that one should be creating a seperate config for a custom pool.